pysilcam/tests/test_process.py

```diff
@@ -19,6 +20,44 @@
 print('MODEL_PATH', MODEL_PATH)
 
 
+@unittest.skipIf((ROOTPATH is None),
+                 "test path not accessible")
+def test_debug_files():
+    '''Testing that the debug images are created'''
+
+    conf_file = os.path.join(ROOTPATH, 'config.ini')
+    conf_file_out = os.path.join(ROOTPATH, 'config_generated.ini')
+    conf = load_config(conf_file)
+
+    data_file = os.path.join(ROOTPATH, 'STN04')
+    conf.set('General', 'loglevel', 'DEBUG')
+    conf.set('General', 'datafile', os.path.join(data_file, 'proc'))
+    conf.set('General', 'logfile', os.path.join(ROOTPATH, 'log.log'))
+    conf.set('ExportParticles', 'outputpath', os.path.join(data_file, 'export'))
+    if MODEL_PATH is not None:
+        conf.set('NNClassify', 'model_path', MODEL_PATH)
+    conf_file_hand = open(conf_file_out, 'w')
+    conf.write(conf_file_hand)
+    conf_file_hand.close()
+
+    # make sure export directory is empty
+    shutil.rmtree(os.path.join(data_file, 'export'), ignore_errors=True)
```

Comment:
Would it be better to use the tempfile library to generate a new randomly named temporary output directory, instead of doing this recursive delete? Such a delete could conceivable go wrong in some edge cases.

Reply:
For instance with

with tempfile.TemporaryDirectory() as tmpdirname:
    <do stuff>

Reply:
Nice idea. I am not completely sure the best way to implement the temporary directory in this case, where the directory that needs cleaning is configured within the config file, which is later read from within the processing function:

conf.set('ExportParticles', 'outputpath', os.path.join(data_file, 'export'))